js/controllers/ideaController.js

```diff
@@ -293,6 +293,21 @@ function($scope, $sce, $http, $filter) {
     popup.classList.toggle("show");
   };
 
+  $scope.showCopiedPopup = function() {
+    $scope.toggleCopiedPopup();
+  };
+
+  $scope.hideCopiedPopup = function() {
+    var popup = document.getElementById("copiedPopup");
+    popup.classList.remove("show");
+  };
+
+  $scope.toggleCopiedPopup = function() {
+    copyIdea();
+    var popup = document.getElementById("copiedPopup");
+    popup.classList.toggle("show");
+  };
+
   $scope.highlight = function() {
```

Comment:
Looks good, do you think you can or myself can add a comment that kind of sections out what the preceding code will do. For example:

```suggestion
# Popup Related Angular JavaScript Function: Toggles popup                                                                                                                                                                                               
  $scope.showCopiedPopup = function() {
    $scope.toggleCopiedPopup();
  };

  $scope.hideCopiedPopup = function() {
    var popup = document.getElementById("copiedPopup");
    popup.classList.remove("show");
  };

  $scope.toggleCopiedPopup = function() {
    copyIdea();
    var popup = document.getElementById("copiedPopup");
    popup.classList.toggle("show");
  };
```

Something to that nature.